jvm-libs/linea/web3j-extensions/src/main/kotlin/linea/web3j/SearchCursor.kt

```diff
@@ -0,0 +1,76 @@
+package linea.web3j
+
+import linea.SearchDirection
+
+/**
+ * SearchCursor is a helper class to iterate over a range of ULong values in a binary search manner.
+ * When search is not provided returns the next unsearched chunk without moving left or right cursors because
+ * no direction is provided, but flags the chunk as searched to not go over again.
+ *
+ * This caveat is because when searching for EthLogs on L1, a blockInterval (represend by chunk here) may not have logs
+ * so the search predicate cannot tell what direction to go next, so we need to flag the chunk as searched and try the
+ * next chunk.
+ */
+class SearchCursor(
```

Comment:
The logic between `Web3JLogsSearcher` and `SearchCursor` is so intertwined, so I'd consider making `SearchCursor` a private class inside `Web3JLogsSearcher`

Reply:
will make it internal, open for the for the tests.